src/SQLStore/PropertyTableInfoFetcher.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace SMW\SQLStore;
+
+use SMW\DataTypeRegistry;
+use SMW\ApplicationFactory;
+use SMW\DIProperty;
+use SMWDataItem as DataItem;
+
+/**
+ * @license GNU GPL v2+
+ * @since 2.2
+ *
+ * @author mwjames
+ */
+class PropertyTableInfoFetcher {
+
+	/**
+	 * Array for keeping property table table data, indexed by table id.
+	 * Access this only by calling getPropertyTables().
+	 *
+	 * @var TableDefinition[]
+	 */
+	private static $propertyTableDefinitions = null;
```

Comment:
Can be done but for this I have to look at Xdebug to ensure we don't start making holes.